### ISSUE TITLE
Error 500 when accessing .../employment-contract page

### ISSUE THREAD

**Issue by Louis-Marie-W** [orangehrm/orangehrm#1698]
### OrangeHRM version

5.4

### Environment details

- OrangeHRM source:   docker file: https://hub.docker.com/r/orangehrm/orangehrm/#!
built via docker-compose.

docker-compose.yml

```
version: '2'

services:
  mariadb:
    user: root
    image: 'bitnami/mariadb:latest'
    environment:
      - MARIADB_DATABASE=orangehrm
      - MARIADB_ROOT_USER=root
      - MARIADB_ROOT_PASSWORD=xxxxxxxxxxxx
      - MARIADB_USER=xxxxxx
      - MARIADB_PASSWORD=xxxxxxxxxxxxxx
    volumes:
      - /var/opt/ohrm-data/mariadb:/bitnami/mariadb
    restart: always
  orangehrm:
    user: root
    image: 'orangehrm/orangehrm:latest'
    environment:
      - MARIADB_HOST=mariadb
      - MARIADB_PORT_NUMBER=3306
      - ORANGEHRM_DATABASE_NAME=orangehrm
      - ORANGEHRM_DATABASE_USER=ohrm
      - ORANGEHRM_DATABASE_PASSWORD=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
      - ORANGEHRM_USERNAME=admin
      - ORANGEHRM_PASSWORD=xxxxxxxxxxxxxx
    hostname: xxxxxxx
    domainname: xxxxxxxxx
    depends_on:
      - mariadb
    restart: always
    volumes:
      - /home/orangehrm/hrm.fair-ezone.com.conf:/etc/apache2/sites-enabled/000-default.conf:ro
      - /etc/ssl/certs/xxxxxxxxxxxxx.crt:/certs/server.crt:ro
      - /etc/ssl/private/xxxxxxxxxxxxxxxxxx.key:/certs/server.key:ro

volumes:
  mariadb_data:
    driver: local
  orangehrm_data:
```

- Platform: Docker version 20.10.7, under Ubuntu 20.04.6 LTS (Focal Fossa)
- PHP version in docker image: 
`PHP 8.1.18 (cli) (built: Apr 14 2023 18:15:35) (NTS)
Zend Engine v4.1.18, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.18, Copyright (c), by Zend Technologies
`
- Database and version: MariaDB 10.11.3
- Web server in docker image:  Apache/2.4.56 (Debian) PHP/8.1.18 OpenSSL/1.1.1n
- Browser: Brave 
[Version 1.51.118 Chromium: 113.0.5672.126 (Official Build) (64-bit)](https://brave.com/latest/)

### Describe the bug

I have just installed a new image of Orangehrm, and migrated from 4.9 to 5.4 without any issue.

Everything seems to work, excepting the page "Job"
https://....../web/index.php/pim/viewJobDetails/empNumber/4
which generate a "GET /web/index.php/api/v2/pim/employees/4/employment-contract" and getting an **error 500**

Apache log:
```
[27/May/2023:12:23:10 +0000] "GET /web/index.php/api/v2/pim/employees/4/employment-contract HTTP/1.1" 500 816 "https://....../web/index.php/pim/viewJobDetails/empNumber/4" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36"
```

/var/www/html/src/log/orangehrm.log (in docker image): (reformatted for lisibility)
```
[2023-05-27T12:22:59.924921+00:00] orangehrm.ERROR:  [] []
[2023-05-27T12:22:59.927064+00:00] orangehrm.ERROR:
 #0 /var/www/html/src/vendor/symfony/event-dispatcher/EventDispatcher.php(230): OrangeHRM\Authentication\Subscriber\AuthenticationSubscriber->onControllerEvent()
 #1 /var/www/html/src/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\Component\EventDispatcher\EventDispatcher->callListeners()
 #2 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(140): Symfony\Component\EventDispatcher\EventDispatcher->dispatch()
 #3 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw()
 #4 /var/www/html/src/lib/framework/Framework.php(157): Symfony\Component\HttpKernel\HttpKernel->handle()
 #5 /var/www/html/web/index.php(42): OrangeHRM\Framework\Framework->handleRequest()
 #6 {main} [] []

[2023-05-27T12:23:10.521450+00:00] orangehrm.CRITICAL: OrangeHRM\Pim\Service\EmploymentContractService::getContractAttachment():
 Return value must be of type ?OrangeHRM\Entity\EmployeeAttachment, OrangeHRM\Pim\Dto\PartialEmployeeAttachment returned [] []

[2023-05-27T12:23:10.521595+00:00] orangehrm.CRITICAL:
 #0 /var/www/html/src/plugins/orangehrmPimPlugin/entity/Decorator/EmpContractDecorator.php(95): OrangeHRM\Pim\Service\EmploymentContractService->getContractAttachment()
 #1 /var/www/html/src/plugins/orangehrmCorePlugin/Api/V2/Serializer/ModelTrait.php(87): OrangeHRM\Entity\Decorator\EmpContractDecorator->getContractAttachment()
 #2 /var/www/html/src/plugins/orangehrmCorePlugin/Api/V2/Serializer/NormalizerTrait.php(80): OrangeHRM\Pim\Api\Model\EmploymentContractModel->toArray()
 #3 /var/www/html/src/plugins/orangehrmCorePlugin/Api/V2/EndpointResourceResult.php(33): OrangeHRM\Core\Api\V2\Serializer\AbstractEndpointResult->normalizeObject()
 #4 /var/www/html/src/plugins/orangehrmCorePlugin/Controller/Rest/V2/GenericRestController.php(208): OrangeHRM\Core\Api\V2\EndpointResourceResult->normalize()
 #5 /var/www/html/src/plugins/orangehrmCorePlugin/Controller/Rest/V2/GenericRestController.php(88): OrangeHRM\Core\Controller\Rest\V2\GenericRestController->getPreparedResponseParamsFromResult()
 #6 /var/www/html/src/plugins/orangehrmCorePlugin/Controller/Rest/V2/AbstractRestController.php(146): OrangeHRM\Core\Controller\Rest\V2\GenericRestController->handleGetRequest()
 #7 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(152): OrangeHRM\Core\Controller\Rest\V2\AbstractRestController->handle()
 #8 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw()
 #9 /var/www/html/src/lib/framework/Framework.php(157): Symfony\Component\HttpKernel\HttpKernel->handle()
 #10 /var/www/html/web/index.php(42): OrangeHRM\Framework\Framework->handleRequest() #11 {main} [] []
```

### To reproduce

It generates the error each time I navigate on Job page, for each employee.

### Expected behavior

It should display the name of the attached file, with a possibility to download it.

### Relevant log output

```shell
Apache log:
[27/May/2023:12:23:10 +0000] "GET /web/index.php/api/v2/pim/employees/4/employment-contract HTTP/1.1" 500 816 "https://....../web/index.php/pim/viewJobDetails/empNumber/4" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36"

/var/www/html/src/log/orangehrm.log (in docker image): (reformatted for lisibility)
[2023-05-27T12:22:59.924921+00:00] orangehrm.ERROR:  [] []
[2023-05-27T12:22:59.927064+00:00] orangehrm.ERROR:
 #0 /var/www/html/src/vendor/symfony/event-dispatcher/EventDispatcher.php(230): OrangeHRM\Authentication\Subscriber\AuthenticationSubscriber->onControllerEvent()
 #1 /var/www/html/src/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\Component\EventDispatcher\EventDispatcher->callListeners()
 #2 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(140): Symfony\Component\EventDispatcher\EventDispatcher->dispatch()
 #3 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw()
 #4 /var/www/html/src/lib/framework/Framework.php(157): Symfony\Component\HttpKernel\HttpKernel->handle()
 #5 /var/www/html/web/index.php(42): OrangeHRM\Framework\Framework->handleRequest()
 #6 {main} [] []

[2023-05-27T12:23:10.521450+00:00] orangehrm.CRITICAL: OrangeHRM\Pim\Service\EmploymentContractService::getContractAttachment():
 Return value must be of type ?OrangeHRM\Entity\EmployeeAttachment, OrangeHRM\Pim\Dto\PartialEmployeeAttachment returned [] []

[2023-05-27T12:23:10.521595+00:00] orangehrm.CRITICAL:
 #0 /var/www/html/src/plugins/orangehrmPimPlugin/entity/Decorator/EmpContractDecorator.php(95): OrangeHRM\Pim\Service\EmploymentContractService->getContractAttachment()
 #1 /var/www/html/src/plugins/orangehrmCorePlugin/Api/V2/Serializer/ModelTrait.php(87): OrangeHRM\Entity\Decorator\EmpContractDecorator->getContractAttachment()
 #2 /var/www/html/src/plugins/orangehrmCorePlugin/Api/V2/Serializer/NormalizerTrait.php(80): OrangeHRM\Pim\Api\Model\EmploymentContractModel->toArray()
 #3 /var/www/html/src/plugins/orangehrmCorePlugin/Api/V2/EndpointResourceResult.php(33): OrangeHRM\Core\Api\V2\Serializer\AbstractEndpointResult->normalizeObject()
 #4 /var/www/html/src/plugins/orangehrmCorePlugin/Controller/Rest/V2/GenericRestController.php(208): OrangeHRM\Core\Api\V2\EndpointResourceResult->normalize()
 #5 /var/www/html/src/plugins/orangehrmCorePlugin/Controller/Rest/V2/GenericRestController.php(88): OrangeHRM\Core\Controller\Rest\V2\GenericRestController->getPreparedResponseParamsFromResult()
 #6 /var/www/html/src/plugins/orangehrmCorePlugin/Controller/Rest/V2/AbstractRestController.php(146): OrangeHRM\Core\Controller\Rest\V2\GenericRestController->handleGetRequest()
 #7 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(152): OrangeHRM\Core\Controller\Rest\V2\AbstractRestController->handle()
 #8 /var/www/html/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw()
 #9 /var/www/html/src/lib/framework/Framework.php(157): Symfony\Component\HttpKernel\HttpKernel->handle()
 #10 /var/www/html/web/index.php(42): OrangeHRM\Framework\Framework->handleRequest() #11 {main} [] []
```

**Comment by RajithaKumara** [comment]
Accepted as a bug. origin #1484

**Comment by RajithaKumara** [comment]
@Louis-Marie-W Thanks for reporting this issue. Please find the patch here, this will be fixed in the next release.

First backup the codebase and the database.
Copy and replace attached files to respective paths in the codebase;

- src/plugins/orangehrmPimPlugin/Api/EmploymentContractAPI.php
- src/plugins/orangehrmPimPlugin/Api/Model/EmploymentContractModel.php
- src/plugins/orangehrmPimPlugin/Service/EmploymentContractService.php
- src/plugins/orangehrmPimPlugin/entity/Decorator/EmpContractDecorator.php

[patch.zip](https://github.com/orangehrm/orangehrm/files/11584949/patch.zip)

Note: This patch was created on OrangeHRM 5.4 version. This might not be compatible with older versions.

**Comment by kavithagurusamy** [comment]
Resolved in 5.5. Upgrade the system to OrangeHRM 5.5 to get the fix.
